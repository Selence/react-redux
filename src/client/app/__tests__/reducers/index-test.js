import React from 'react';
import Redux from 'redux';
import renderer from 'react-test-renderer';
import Reducers from '../../reducers/index'

const mockRedux = Redux;
jest.mock(
    'redux',
    () => ({
        ...mockRedux,
        combineReducers: (props) => (JSON.stringify(props))
    })
);

jest.mock(
    '../../reducers/CounterReducer',
    () => ({
        counter: {type: 'ACTION_COUNTER'}
    })
);

describe('Reducers Index', () => {
    const renderComponent = () => (
        renderer.create(<Reducers />).toJSON()
    );

    describe('when the reducers are called', () => {
        it('they renders correctly', () => {
            expect(renderComponent()).toMatchSnapshot();
        });
    })
});
