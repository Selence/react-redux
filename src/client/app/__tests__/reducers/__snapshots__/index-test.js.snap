// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers Index when the reducers are called they renders correctly 1`] = `<{"counter":{"type":"ACTION_COUNTER"}} />`;
